electron/syntheticsGenerator.ts

```diff
@@ -212,27 +223,25 @@ export class SyntheticsGenerator extends PlaywrightGenerator.JavaScriptLanguageG
     // Popup signals.
     if (signals.popup) formatter.add(`${pageAlias}.waitForEvent('popup'),`);
 
-    // Navigation signal.
-    if (signals.waitForNavigation) formatter.add(`${pageAlias}.waitForNavigation(),`);
-
     // Download signals.
     if (signals.download) formatter.add(`${pageAlias}.waitForEvent('download'),`);
 
-    const prefix = signals.popup || signals.waitForNavigation || signals.download ? '' : 'await ';
+    const prefix = signals.popup || signals.download ? '' : 'await ';
     const actionCall = super._generateActionCall(action);
     // Add assertion from Synthetics.
     const isAssert = action.name === 'assert' && action.isAssert;
 
     if (!isAssert) {
-      const suffix = signals.waitForNavigation || emitPromiseAll ? '' : ';';
+      const suffix = emitPromiseAll ? '' : ';';
       formatter.add(`${prefix}${subject}.${actionCall}${suffix}`);
 
       if (emitPromiseAll) {
         formatter.add(']);');
-      } else if (signals.assertNavigation) {
-        formatter.add(
-          `  expect(${pageAlias}.url()).toBe(${quote(signals.assertNavigation.url ?? '')});`
-        );
+      } else if (signals.assertNavigation?.url) {
+        // original code uses the full URL including query strings which often contains randomly generated value(cashbusting)
+        // when replaying the script it will fail so we will check only baseurl and path
+        const url = new URL(signals.assertNavigation.url);
```

Comment:
Does it parse query by default? can you disable if it does?

Reply:
It does parse query by default - and it doesn't seem to have an option for disabling it. Can I ask what's your concern about query?